distage/distage-framework-docker/src/main/scala/izumi/distage/docker/DockerContainer.scala

```diff
@@ -178,55 +183,89 @@ object DockerContainer {
       }
     }
 
-    private[this] def runReused(ports: Seq[PortDecl]): F[DockerContainer[T]] = {
-      for {
-        containers <- DIEffect[F].maybeSuspend {
-          // FIXME: temporary hack to allow missing containers to skip tests (happens when both DockerWrapper & integration check that depends on Docker.Container are memoized)
-          try {
-            client
-              .listContainersCmd()
-              .withAncestorFilter(List(config.image).asJava)
-              .withStatusFilter(List("running").asJava)
-              .exec()
-          } catch {
-            case c: Throwable =>
-              throw new IntegrationCheckException(Seq(ResourceCheck.ResourceUnavailable(c.getMessage, Some(c))))
-          }
-        }.map(_.asScala.toList.sortBy(_.getId))
-
-        candidates = {
-          val portSet = ports.map(_.port).toSet
-          containers.iterator.flatMap {
-            c =>
-              val inspection = client.inspectContainerCmd(c.getId).exec()
-              mapContainerPorts(inspection) match {
-                case Left(value) =>
-                  logger.info(s"Container ${c.getId} missing ports $value so will not be reused")
-                  Seq.empty
-                case Right(value) =>
-                  Seq((c, inspection, value))
-              }
-          }.find {
-            case (_, _, eports) =>
-              portSet.diff(eports.keySet).isEmpty
-          }
+    private[this] def withLocalMutex[E](waitForMillis: Int = 1000, maxAttempts: Int = 10)(eff: F[E]): F[E] = {
+      @tailrec
+      def acquireExclusiveLock(chanel: FileChannel, attempts: Int = 0): Option[FileLock] = {
+        logger.debug(s"Attempt ${attempts -> "num"} to acquire lock for ${config.image}.")
+        (try {
+          Option(chanel.tryLock())
+        } catch {
+          case _: OverlappingFileLockException => None
+        }) match {
+          case v@Some(_) => v
+          case None if attempts < maxAttempts =>
+            Thread.sleep(waitForMillis)
```

Comment:
@Caparow I think with wait times above 50ms we should really use `DIEffectAsync.sleep` instead of threadDelay to avoid hogging up threads

Reply:
done